services/web-server/src/servers/createSubscriptionServer.js

```diff
@@ -2,18 +2,75 @@ import { SubscriptionServer } from 'subscriptions-transport-ws';
 import { execute, subscribe } from 'graphql';
 import credentials from './credentials.js';
 import formatError from './formatError.js';
+import scopeUtils from 'taskcluster-lib-scopes';
+import { decryptToken } from './decryptToken.js';
+import { ErrorReply } from 'taskcluster-lib-api';
+
+// TODO: Check for expiration of access token when the websocket connection is active
+export default ({ cfg, server, schema, context, path, authFactory }) => {
+
+  const timeoutMap = new WeakMap();
 
-export default ({ server, schema, context, path }) =>
   SubscriptionServer.create(
     {
       schema,
       execute,
       subscribe,
-      onConnect(params, socket) {
-        return new Promise(resolve => {
-          credentials()(socket.upgradeReq, {}, resolve);
+      async onConnect(params, socket) {
+
+        const disconnectTimeout = setTimeout(() => {
+          if (socket && socket.readyState !== socket.CLOSING && socket.readyState !== socket.CLOSED) {
+            socket.close();
+          }
+        }, cfg.server.socketAliveTimeoutMilliSeconds);
+        timeoutMap.set(socket, disconnectTimeout);
+
+        return new Promise((resolve, reject) => {
+          credentials()(socket.upgradeReq, {}, async () => {
+            try {
+
+              const credentials = params?.Authorization ? decryptToken(params.Authorization) : null;
+
+              const authClient = authFactory(credentials);
```

Comment:
I think you need to pass object, since factory expects it in this format, or just change to single argument
```suggestion
              const authClient = authFactory({ credentials });
```

Reply:
Resolved 👍